utilities/make_substructure_dict/_make_chemical_substructures.py

```diff
@@ -33,3 +33,151 @@
+    return max([atom.GetAtomMapNum() for atom in rdmol.GetAtoms()])
+
+def _fill_out_query(rdmol):
+    current_map_num = 1 + _get_maximum_map_num(rdmol) # do not change existing atom map numbers! 
+    for atom in rdmol.GetAtoms():
+        if atom.GetAtomicNum() > 0:
+            if atom.GetAtomMapNum() == 0:
+                atom.SetAtomMapNum(current_map_num)
+                current_map_num += 1
+            atom.SetChiralTag(Chem.rdchem.ChiralType.CHI_UNSPECIFIED)
+            a_num = atom.GetAtomicNum()
+            D_num = len([0 for _ in atom.GetBonds()])
+            F_num = atom.GetFormalCharge()
+            query_string = f"[#{a_num}D{D_num}{F_num:+}:{current_map_num}]"
+            query = Chem.AtomFromSmarts(query_string)
+            atom.SetQuery(query)
+        elif atom.GetAtomicNum() == 0:
+            atom.SetAtomMapNum(current_map_num)
+            current_map_num += 1
+            query_string = f"[*:{current_map_num}]"
+            query = Chem.AtomFromSmarts(query_string)
+            atom.SetQuery(query)
+        else:
+            raise Exception
+    for bond in rdmol.GetBonds():
+        if bond.GetIsAromatic():
+            query = Chem.BondFromSmarts(":")
+            bond.SetQuery(query)
+        elif bond.GetBondType() == Chem.rdchem.BondType.SINGLE:
+            query = Chem.BondFromSmarts("-")
+            bond.SetQuery(query)
+        
+        
+    return rdmol
+
+def get_atom_with_map_num(rdmol, map_num):
+    for atom in rdmol.GetAtoms():
+        if atom.GetAtomMapNum() == map_num:
+            return atom
+    return None
+
+def clear_query(rdmol):
+    for atom in rdmol.GetAtoms():
+        atom.SetAtomMapNum(0)
+        n = atom.GetAtomicNum()
+        c = atom.GetFormalCharge()
+        query_string = f"[#{n}{c:+}]"
+        query = Chem.AtomFromSmarts(query_string)
+        atom.SetQuery(query)
+
+substructure_file_path = get_data_file_path(
+    "proteins/aa_residues_substructures_explicit_bond_orders_with_caps.json"
+)
+
+with open(substructure_file_path, "r") as subfile:
+    substructure_dictionary = json.load(
+        subfile, object_pairs_hook=OrderedDict
+    )
+
+new_subs_dict = OrderedDict()
+amino_acid_subs = [(Chem.MolFromSmarts("[N+:1](-[H])(-[H])1[C@@:2]([C:3](=[O:4]))([H:9])[C:5]([H:10])([H:11])[C:6]([H:12])([H:13])[C:7]1([H:14])[H:15]"), [], [], 4),
+                   (Chem.MolFromSmarts("[N+0:1](-[H])1[C@@:2]([C:3](=[O:4]))([H:9])[C:5]([H:10])([H:11])[C:6]([H:12])([H:13])[C:7]1([H:14])[H:15]"), [], [], 3),
+                   (Chem.MolFromSmarts("[N+1:1](-[H])1[C@@:2]([C:3](=[O:4]))([H:9])[C:5]([H:10])([H:11])[C:6]([H:12])([H:13])[C:7]1([H:14])[H:15]"), [0], [], 3),
+                   (Chem.MolFromSmarts("[N+0:1]1[C@@:2]([C:3](=[O:4]))([H:9])[C:5]([H:10])([H:11])[C:6]([H:12])([H:13])[C:7]1([H:14])[H:15]"), [0], [], 2),
+                   (Chem.MolFromSmarts("[N+](-[H])(-[H])(-[H])-[C@:2]([C:3](=[O:4]))([C:5]([S+0:6]([H]))([H:9])[H:10])[H:8]"), [], [], 5),
```

Comment:
I'm not sure I see the purpose of this section. Is it redundant with [this](https://github.com/openforcefield/openff-toolkit/blob/f84ca0e1955f151cd9c842fbd9bb144f486ef29c/utilities/make_substructure_dict/_cif_to_substructure_dict.py#L610) or does it serve a different purpose?

Reply:
This section adds wildtype atoms where needed to satisfy the new spec, so I don't think it is redundant. However, I might see if this code can maybe run further up and will add a bit more documentation here to explain.